src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt

```diff
@@ -744,6 +744,264 @@ class TasksTest : IntegrationTestBase() {
+
+            val numAllocatedPlacementApplications = 2
+            val numUnallocatedPlacementApplications = 3
+            val numAllocatableAssessment = 4
+            val numUnallocatableAssessment = 5
+            val numAllocatedPlacementRequests = 6
+            val numUnallocatedPlacementRequests = 7
+
+            repeat(numAllocatedPlacementApplications) {
+              `Given a Placement Application`(
+                createdByUser = user,
+                allocatedToUser = user,
+                schema = approvedPremisesPlacementApplicationJsonSchemaEntityFactory.produceAndPersist {
+                  withPermissiveSchema()
+                },
+                crn = offenderDetails.otherIds.crn,
+                submittedAt = OffsetDateTime.now(),
+              )
+            }
+
+            repeat(numUnallocatedPlacementApplications) {
+              `Given a Placement Application`(
+                createdByUser = user,
+                schema = approvedPremisesPlacementApplicationJsonSchemaEntityFactory.produceAndPersist {
+                  withPermissiveSchema()
+                },
+                crn = offenderDetails.otherIds.crn,
+                submittedAt = OffsetDateTime.now(),
+              )
+            }
+
+            repeat(numAllocatableAssessment) {
+              `Given an Assessment for Approved Premises`(
+                allocatedToUser = otherUser,
+                createdByUser = otherUser,
+                crn = offenderDetails.otherIds.crn,
+              )
+            }
+
+            repeat(numUnallocatableAssessment) {
+              `Given an Assessment for Approved Premises`(
+                null,
+                createdByUser = otherUser,
+                crn = offenderDetails.otherIds.crn,
+              )
+            }
+
+            repeat(numAllocatedPlacementRequests) {
+              `Given a Placement Request`(
+                placementRequestAllocatedTo = otherUser,
+                assessmentAllocatedTo = otherUser,
+                createdByUser = user,
+                crn = offenderDetails.otherIds.crn,
+              )
+            }
+
+            repeat(numUnallocatedPlacementRequests) {
+              `Given a Placement Request`(
+                null,
+                assessmentAllocatedTo = otherUser,
+                createdByUser = user,
+                crn = offenderDetails.otherIds.crn,
+              )
+            }
+
+            webTestClient.get()
+              .uri("/tasks/reallocatable?page=2")
+              .header("Authorization", "Bearer $jwt")
+              .exchange()
+              .expectStatus()
+              .isOk
+              .expectHeader().valueEquals("X-Pagination-CurrentPage", 2)
+              .expectHeader().valueEquals("X-Pagination-TotalPages", 3)
+              .expectHeader().valueEquals("X-Pagination-TotalResults", 27)
+              .expectHeader().valueEquals("X-Pagination-PageSize", 10)
```

Comment:
Would be good for this to be a sum of  numAllocatedPlacementApplications,  numUnallocatedPlacementApplications etc etc to save this being a magic number